contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -33,56 +96,161 @@ contract ProtocolShareReserve is Ownable2StepUpgradeable, ExponentialNoError, Re
+        //because prime relies on getUnreleasedFunds and it's return value may change after config update
+        _accrueAndReleaseFundsToPrime();
+
+        uint256 total = 0;
+        bool updated = false;
+        for (uint i = 0; i < distributionTargets.length; i++) {
+            DistributionConfig storage _config = distributionTargets[i];
+            if (_config.schema == config.schema) {
+                if (config.destination == _config.destination) {
+                    total += config.percentage;
+                    _config.percentage = config.percentage;
+                    updated = true;
+                } else {
+                    total += _config.percentage;
+                }
+            }
+        }
+
+        if (updated == false) {
+            total += config.percentage;
+            distributionTargets.push(config);
+        }
+
+        require(total <= MAX_PERCENT, "ProtocolShareReserve: Percentage must be between 0 and 100");
+    }
+
+    /**
+     * @dev Used to find out the amount of funds that's going to be released when release funds is called.
+     * @param comptroller the comptroller address of the pool
+     * @param schema the schema of the distribution target
+     * @param destination the destination address of the distribution target
+     * @param asset the asset address which will be released
+     */
+    function getUnreleasedFunds(
+        address comptroller,
+        Schema schema,
+        address destination,
+        address asset
+    ) external view returns (uint256) {
+        for (uint i = 0; i < distributionTargets.length; i++) {
+            DistributionConfig storage _config = distributionTargets[i];
+            if (_config.schema == schema && _config.destination == destination) {
+                uint256 total = assetsReserves[comptroller][asset][schema];
+                return (total * _config.percentage) / MAX_PERCENT;
+            }
+        }
     }
 
     /**
      * @dev Release funds
-     * @param asset  Asset to be released
-     * @param amount Amount to release
-     * @return Number of total released tokens
+     * @param assets assets to be released to distribution targets
      */
-    function releaseFunds(address comptroller, address asset, uint256 amount) external returns (uint256) {
-        require(asset != address(0), "ProtocolShareReserve: Asset address invalid");
-        require(amount <= poolsAssetsReserves[comptroller][asset], "ProtocolShareReserve: Insufficient pool balance");
+    function releaseFunds(address comptroller, address[] memory assets) external {
+        for (uint i = 0; i < assets.length; i++) {
+            _releaseFund(comptroller, assets[i]);
+        }
+    }
+
+    function _releaseFund(address comptroller, address asset) internal {
+        uint256 schemaOneBalance = assetsReserves[comptroller][asset][Schema.ONE];
+        uint256 schemaTwoBalance = assetsReserves[comptroller][asset][Schema.TWO];
```

Comment:
Do we have to take care of the case where there isn't a market for the asset in the provided comptroller?